src/shaka_scheme/system/gc/GCList.cpp

```diff
@@ -4,67 +4,78 @@
 
 #include "shaka_scheme/system/gc/GCData.hpp"
 #include "shaka_scheme/system/gc/GCList.hpp"
+#include <utility>
 
 namespace shaka {
+    namespace gc {
 
-    GCList::GCList() {
-        this->size = 0;
-        this->head = nullptr;
-    }
+        GCList::GCList() {
+            this->size = 0;
+            this->head = nullptr;
+        }
 
-    GCList::~GCList() {
-        GCData *conductor = this->head;
+        GCList::~GCList() {
+            GCData *conductor = this->head;
 
-        while(conductor != nullptr) {
-            GCData *temp = conductor;
-            conductor = conductor->get_next();
-            delete temp;
-            this->size--;
+            while (conductor != nullptr) {
+                GCData *temp = conductor;
+                conductor = conductor->get_next();
+                delete temp;
+                this->size--;
```

Comment:
Why decrement here? You are deleting the entire list, so it'll be destroyed anyways.

Reply:
Yeah you're probably right about that. I added that it in mainly because I wanted to insure the list decreased as each GCData object was removed from the list and thus prevent memory leak. If you don't think memory leak is an issue here, I will remove that decrement counter. @VermillionAzure